content/recipes/swagger.md

```diff
@@ -348,52 +412,219 @@ Let's open the browser and verify the generated `Cat` model:
+| `@ApiExtraModels()`      | Method / Controller |
+| `@ApiBody()`             | Method              |
+| `@ApiParam()`            | Method              |
+| `@ApiQuery()`            | Method              |
+| `@ApiHeader()`           | Method / Controller |
+| `@ApiExcludeEndpoint()`  | Method              |
+| `@ApiTags()`             | Method / Controller |
+| `@ApiProperty()`         | Model               |
+| `@ApiPropertyOptional()` | Model               |
+| `@ApiHideProperty()`     | Model               |
+
+#### Plugin
+
+TypeScript metadata reflaction system has several limitations, which make it impossible to, for instance, determine what properties a class consist of or recognize whether a given property is optional or required. However, some of these constraints can be addressed at the compilation time. Hence, we created a plugin that enhances TypeScript compilation process to reduce the boilerplate code.
+
+Swagger plugin will automatically:
+
+- annotate all DTO properties with `@ApiProperty` unless `@ApiHideProperty` is used
+- set `required` property depending on the question mark (e.g. `name?: string` = `required: false`)
+- set `type` or `enum` property depending on the type (supports arrays as well)
+- set `default` property based on the assigned default value
+- set several validation rules based on `class-validator` decorators (if `classValidatorShim` set to `true`)
+- add response decorator to every endpoint with a proper status and `type` (response model)
+
+In order to enable plugin, simply open `nest-cli.json` (if you use [Nest CLI](/cli/overview)) and add `plugins` configuration:
+
+```javascript
+{
+  "collection": "@nestjs/schematics",
+  "sourceRoot": "src",
+  "compilerOptions": {
+    "plugins": [
+      {
+        "name": "@nestjs/swagger/plugin",
+        "options": {}
+      }
+    ]
+  }
+}
+```
+
+You can use `options` to customize the behavior of the plugin. The `options` has to fulfill the following interface:
+
+```typescript
+export interface PluginOptions {
+  dtoFileNameSuffix?: string;
+  controllerFileNameSuffix?: string;
+  classValidatorShim?: boolean;
+}
+```
+
+<table>
+  <tr>
+    <th>Option</th>
+    <th>Default</th>
+    <th>Description</th>
+  </tr>
+  <tr>
+    <td><code>dtoFileNameSuffix</code></td>
+    <td><code>.dto.ts</code></td>
+    <td>DTO (Data Transfer Object) files suffix</td>
+  </tr>
+  <tr>
+    <td><code>controllerFileNameSuffix</code></td>
+    <td><code>.controller.ts</code></td>
+    <td>Controller files suffix</td>
+  </tr>
+  <tr>
+    <td><code>classValidatorShim</code></td>
+    <td><code>false</code></td>
+    <td>If set to true, the module will reuse <code>class-validator</code> validation decorators (e.g. <code>@Max(10)</code> will add <code>max: 10</code> to schema definition) </td>
+  </tr>
+</table>
+
+If you don't use a CLI but rather have a custom `webpack` configuration, you can still use this plugin in combination with `ts-loader`:
```

Comment:
If you don't use a CLI but rather have a custom `webpack` configuration, you can still use this plugin in combination with `ts-loader`:

->

If you don't use the Nest CLI but instead have a custom `webpack` configuration, you can use this plugin in combination with `ts-loader`: